crates/services/block_aggregator_api/src/blocks/importer_and_db_source/sync_service.rs

```diff
@@ -0,0 +1,171 @@
+use crate::blocks::{
+    BlockSourceEvent,
+    importer_and_db_source::BlockSerializer,
+};
+use fuel_core_services::{
+    RunnableService,
+    RunnableTask,
+    StateWatcher,
+    TaskNextAction,
+    try_or_continue,
+    try_or_stop,
+};
+use fuel_core_storage::{
+    self,
+    StorageInspect,
+    tables::{
+        FuelBlocks,
+        Transactions,
+    },
+};
+use fuel_core_types::{
+    fuel_tx::{
+        Transaction,
+        TxId,
+    },
+    fuel_types::BlockHeight,
+};
+use tokio::sync::mpsc::Sender;
+
+pub struct SyncTask<Serializer, DB> {
+    serializer: Serializer,
+    block_return_sender: Sender<BlockSourceEvent>,
+    db: DB,
+    next_height: BlockHeight,
+    maybe_stop_height: Option<BlockHeight>,
+    new_ending_height: tokio::sync::oneshot::Receiver<BlockHeight>,
+}
+
+impl<Serializer, DB, E> SyncTask<Serializer, DB>
+where
+    Serializer: BlockSerializer + Send,
+    DB: StorageInspect<FuelBlocks, Error = E> + Send + 'static,
+    DB: StorageInspect<Transactions, Error = E> + Send + 'static,
+    E: std::fmt::Debug + Send,
+{
+    pub fn new(
+        serializer: Serializer,
+        block_return: Sender<BlockSourceEvent>,
+        db: DB,
+        db_starting_height: BlockHeight,
+        db_ending_height: Option<BlockHeight>,
+        new_ending_height: tokio::sync::oneshot::Receiver<BlockHeight>,
+    ) -> Self {
+        Self {
+            serializer,
+            block_return_sender: block_return,
+            db,
+            next_height: db_starting_height,
+            maybe_stop_height: db_ending_height,
+            new_ending_height,
+        }
+    }
+
+    async fn maybe_update_stop_height(&mut self) {
+        if let Ok(last_height) = self.new_ending_height.try_recv() {
+            tracing::info!("updating last height to {}", last_height);
+            self.maybe_stop_height = Some(last_height);
+        }
+    }
+
+    fn get_block(
```

Comment:
I would like to emphasize that we also require receipts from this service. We should be able to return a block with transactions and receipts.